mage_ai/data_cleaner/column_type_detector.py

```diff
@@ -54,143 +57,114 @@ def get_mismatched_row_count(series, column_type):
-                        binary_feature_names.append(col_name)
-                    else:
-                        category_feature_names.append(col_name)
-                elif len(df[col_name].unique()) <= 2:
-                    binary_feature_names.append(col_name)
-                else:
-                    df_sub = df_sub.astype(str)
-                    incorrect_emails = len(
-                        df_sub[~df_sub.str.contains(REGEX_EMAIL)].index,
-                    )
-                    warnings.filterwarnings('ignore', 'This pattern has match groups')
-                    incorrect_phone_numbers = len(
-                        df_sub[~df_sub.str.contains(REGEX_PHONE_NUMBER)].index,
-                    )
-                    incorrect_zip_codes = len(
-                        df_sub[~df_sub.str.contains(REGEX_ZIP_CODE)].index,
-                    )
+    return mdtype
 
-                    if all(df_sub.str.contains(REGEX_INTEGER)):
-                        integer_feature_names.append(col_name)
-                    elif all(df_sub.str.contains(REGEX_NUMBER)):
-                        float_feature_names.append(col_name)
-                    elif incorrect_emails / len(df_sub.index) <= 0.99:
-                        email_features.append(col_name)
-                    elif incorrect_phone_numbers / len(df_sub.index) <= 0.99:
-                        phone_number_feature_names.append(col_name)
-                    elif incorrect_zip_codes / len(df_sub.index) <= 0.99:
-                        zip_code_feature_names.append(col_name)
-                    else:
-                        non_number_feature_names.append(col_name)
-        elif col_type == 'bool':
-            binary_feature_names.append(col_name)
-        elif np.issubdtype(col_type, np.floating):
-            float_feature_names.append(col_name)
-        elif np.issubdtype(col_type, np.integer):
-            df_sub = df[col_name].copy()
-            df_sub = df_sub.dropna()
-            if df_sub.min() >= 100 and df_sub.max() <= 99999 and 'zip' in col_name.lower():
-                zip_code_feature_names.append(col_name)
-            else:
-                integer_feature_names.append(col_name)
+def infer_object_type(series, kwargs):
+    clean_series = series.apply(lambda x: x.strip(' \'\"') if type(x) is str else x)
+    clean_series = clean_whitespace(clean_series)
+    clean_series = clean_series.dropna()
 
-    number_feature_names = float_feature_names + integer_feature_names
-    binary_feature_names += \
-        [col for col in number_feature_names if df[col].nunique(dropna=False) == 2]
-    binary_feature_names += \
-        [col for col in non_number_feature_names if df[col].nunique(dropna=False) == 2]
-    float_feature_names = [col for col in float_feature_names if col not in binary_feature_names]
-    integer_feature_names = \
-        [col for col in integer_feature_names if col not in binary_feature_names]
+    series_unique = series.nunique(dropna=False)
```

Comment:
nit: series_nunique and clean_series_nunique

Reply:
Fixed